diff --git a/Engine/Source/Programs/UnrealBuildTool/ProjectFiles/Xcode/XcodeProject.File.cs b/Engine/Source/Programs/UnrealBuildTool/ProjectFiles/Xcode/XcodeProject.File.cs
index 8296d8f4a..9d1138087 100644
--- a/Engine/Source/Programs/UnrealBuildTool/ProjectFiles/Xcode/XcodeProject.File.cs
+++ b/Engine/Source/Programs/UnrealBuildTool/ProjectFiles/Xcode/XcodeProject.File.cs
@@ -221,7 +221,7 @@ namespace UnrealBuildTool.XcodeProjectXcconfig
 
 		public void SetUProjectLocation(FileReference? UProjectLocation)
 		{
-			RootDirectory = UProjectLocation == null || UProjectLocation.ContainsName("Programs", 0) ? (Unreal.EngineDirectory) : UProjectLocation!.Directory;
+			RootDirectory = UProjectLocation == null || (UProjectLocation.ContainsName(Unreal.EngineDirectory.FullName, 0) && UProjectLocation.ContainsName("Programs", 0)) ? (Unreal.EngineDirectory) : UProjectLocation!.Directory;
 		}
 
 		public void ProcessFile(XcodeSourceFile File, bool bIsForBuild, bool bIsFolder, string? GroupName = null, string ExtraResourceSettings = "", Dictionary<DirectoryReference, int>? SourceToBuildFileMap = null)
